src/mono/wasm/debugger/BrowserDebugProxy/MemberReferenceResolver.cs

```diff
@@ -119,13 +119,38 @@ public async Task<JObject> TryToRunOnLoadedClasses(string varName, CancellationT
                     }
                 }
                 var store = await proxy.LoadStore(sessionId, token);
-                foreach (var asm in store.assemblies)
+                var methodInfo = ctx.CallStack.FirstOrDefault(s => s.Id == scopeId)?.Method?.Info;
+                var classNameToFindWithNamespace =
+                    string.IsNullOrEmpty(methodInfo?.TypeInfo?.Namespace) ?
+                    classNameToFind :
+                    methodInfo.TypeInfo.Namespace + "." + classNameToFind;
+
+                var searchResult = await TryFindNameInAssembly(store.assemblies, classNameToFindWithNamespace);
+                if (searchResult == null)
+                    searchResult = await TryFindNameInAssembly(store.assemblies, classNameToFind);
+                if (searchResult != null)
+                    typeId = (int)searchResult;
+
+                async Task<Tuple<bool, int?>> TryGetTypeIdFromName(string typeName, AssemblyInfo assembly)
                 {
-                    var type = asm.GetTypeByName(classNameToFind);
-                    if (type != null)
+                    int typeId;
+                    var type = assembly.GetTypeByName(typeName);
+                    if (type == null)
+                        return new Tuple<bool, int?>(false, null);
+
+                    typeId = await sdbHelper.GetTypeIdFromToken(sessionId, assembly.DebugId, type.Token, token);
+                    return new Tuple<bool, int?>(true, typeId); ;
+                }
+
+                async Task<int?> TryFindNameInAssembly(List<AssemblyInfo> assemblies, string name)
+                {
+                    foreach (var asm in assemblies)
                     {
-                        typeId = await sdbHelper.GetTypeIdFromToken(sessionId, asm.DebugId, type.Token, token);
+                        var searchResult = await TryGetTypeIdFromName(name, asm);
+                        if (searchResult.Item1)
+                            return searchResult.Item2;
                     }
```

Comment:
You can [deconstruct ](https://docs.microsoft.com/en-us/dotnet/csharp/fundamentals/functional/deconstruct)the return value:

`(bool result, int? typeId) = await TryGetTypeIdFromName(name, asm);`

Reply:
though if you are returning `int?`, then the check can simple by `typeId != null`